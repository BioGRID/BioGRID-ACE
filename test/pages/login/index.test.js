
// LIBRARIES
import Vuetify from 'vuetify'

// COMPONENTS
import index from '@/pages/login/index.vue'

// UTILITIES
import { mount, createLocalVue } from '@vue/test-utils'

const localVue = createLocalVue()

let wrapper
let emailField
let passwordField

beforeEach(async () => {
    const vuetify = new Vuetify()
    wrapper = mount(index, {
        localVue,
        vuetify
    })

    emailField = await wrapper.find('input[name="email"]')
    passwordField = await wrapper.find('input[name="password"]')
})

afterEach(() => {
    wrapper.destroy()
})

describe('pages/login/index.vue', () => {
    test('renders', () => {
        expect(wrapper.exists()).toBe(true)
    })
    test('renders properly', () => {
        expect(wrapper.html()).toMatchSnapshot()
    })
    test('email field has errors with invalid email', () => {
        emailField.setValue('test')
        expect(wrapper.vm.emailErrors.length).toBeGreaterThan(0)
    })
    test('email field has no errors with valid email', () => {
        emailField.setValue('test@gmail.com')
        expect(wrapper.vm.emailErrors.length).toBe(0)
    })
    test('email field has error if originally valid but then erased', () => {
        emailField.setValue('test@gmail.com')
        expect(wrapper.vm.emailErrors.length).toBe(0)
        emailField.setValue('')
        expect(wrapper.vm.emailErrors.length).toBeGreaterThan(0)
    })
    test('password field has error if originally valid but then erased', () => {
        passwordField.setValue('supersecret')
        expect(wrapper.vm.passwordErrors.length).toBe(0)
        passwordField.setValue('')
        expect(wrapper.vm.passwordErrors.length).toBeGreaterThan(0)
    })
    test('login button is disabled if any field errors exist', () => {
        expect(wrapper.vm.isInvalid).toBeTruthy()
        emailField.setValue('test@gmail.com')
        expect(wrapper.vm.isInvalid).toBeTruthy()
        emailField.setValue('')
        expect(wrapper.vm.isInvalid).toBeTruthy()
        passwordField.setValue('supersecret')
        expect(wrapper.vm.isInvalid).toBeTruthy()
        emailField.setValue('test@gmail.com')
        expect(wrapper.vm.isInvalid).toBeFalsy()
    })
    test('enter key on email or password field triggers login', async () => {
        const spyLogin = jest.spyOn(wrapper.vm, 'login')
        await emailField.trigger('keyup.enter')
        expect(spyLogin).toHaveBeenCalled()
        await passwordField.trigger('keyup.enter')
        expect(spyLogin).toHaveBeenCalled()
    })
    test('button click triggers login', async () => {
        const spyLogin = jest.spyOn(wrapper.vm, 'login')
        emailField.setValue('test@gmail.com')
        passwordField.setValue('supersecret')
        const button = await wrapper.find('button')
        button.trigger('click')
        expect(spyLogin).toHaveBeenCalled()
    })
})
