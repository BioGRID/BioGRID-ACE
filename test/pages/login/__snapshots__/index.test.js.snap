// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`pages/login/index.vue renders properly 1`] = `
"<div class=\\"container fill-height\\">
  <div class=\\"layout row wrap align-center\\">
    <div class=\\"flex text-center\\">
      <div class=\\"mx-auto v-card v-card--raised v-sheet v-sheet--outlined theme--light\\" style=\\"height: 320px; width: 800px;\\">
        <div class=\\"container pa-0\\">
          <div class=\\"row no-gutters\\">
            <div class=\\"pr-0 pt-0 pb-0 col-sm-6 col\\">
              <div class=\\"mx-auto ml-0 v-card v-card--flat v-sheet theme--light\\">
                <div class=\\"v-image v-responsive pa-0 theme--light\\" style=\\"height: 319px;\\">
                  <transition-stub name=\\"fade-transition\\" mode=\\"in-out\\">
                    <div class=\\"v-image__image v-image__image--preload v-image__image--cover\\" style=\\"background-position: center center;\\"></div>
                  </transition-stub>
                  <div class=\\"v-responsive__content\\"></div>
                </div>
              </div>
            </div>
            <div class=\\"col-sm-6 col\\">
              <div class=\\"text-left ml-3 mt-3 mr-3 v-card v-card--flat v-sheet theme--light\\" style=\\"height: 295px;\\">
                <div class=\\"row\\">
                  <div class=\\"col-sm-12 col\\">
                    <div class=\\"v-image v-responsive theme--light\\" style=\\"width: 180px;\\">
                      <transition-stub name=\\"fade-transition\\" mode=\\"in-out\\">
                        <div class=\\"v-image__image v-image__image--preload v-image__image--contain\\" style=\\"background-position: center center;\\"></div>
                      </transition-stub>
                      <div class=\\"v-responsive__content\\"></div>
                    </div>
                    <p class=\\"subtitle-2 mt-1\\">
                      Welcome Back, please login to proceed...
                    </p>
                    <form novalidate=\\"novalidate\\" class=\\"v-form\\">
                      <div class=\\"v-input login-field theme--light v-text-field\\">
                        <div class=\\"v-input__control\\">
                          <div class=\\"v-input__slot\\">
                            <div class=\\"v-input__prepend-inner\\">
                              <div class=\\"v-input__icon v-input__icon--prepend-inner\\"><i aria-hidden=\\"true\\" class=\\"v-icon notranslate mdi mdi-clipboard-account theme--light\\"></i></div>
                            </div>
                            <div class=\\"v-text-field__slot\\"><label for=\\"input-43\\" class=\\"v-label theme--light\\" style=\\"left: 0px; position: absolute;\\">Email</label><input name=\\"email\\" required=\\"required\\" id=\\"input-43\\" type=\\"text\\"></div>
                          </div>
                          <div class=\\"v-text-field__details\\">
                            <div class=\\"v-messages theme--light\\">
                              <transition-group-stub tag=\\"div\\" name=\\"message-transition\\" class=\\"v-messages__wrapper\\"></transition-group-stub>
                            </div>
                          </div>
                        </div>
                      </div>
                      <div class=\\"v-input login-field theme--light v-text-field\\">
                        <div class=\\"v-input__control\\">
                          <div class=\\"v-input__slot\\">
                            <div class=\\"v-input__prepend-inner\\">
                              <div class=\\"v-input__icon v-input__icon--prepend-inner\\"><i aria-hidden=\\"true\\" class=\\"v-icon notranslate mdi mdi-lock-question theme--light\\"></i></div>
                            </div>
                            <div class=\\"v-text-field__slot\\"><label for=\\"input-47\\" class=\\"v-label theme--light\\" style=\\"left: 0px; position: absolute;\\">Password</label><input name=\\"password\\" required=\\"required\\" id=\\"input-47\\" type=\\"password\\"></div>
                          </div>
                          <div class=\\"v-text-field__details\\">
                            <div class=\\"v-messages theme--light\\">
                              <transition-group-stub tag=\\"div\\" name=\\"message-transition\\" class=\\"v-messages__wrapper\\"></transition-group-stub>
                            </div>
                          </div>
                        </div>
                      </div> <button type=\\"button\\" disabled=\\"disabled\\" class=\\"mt-4 v-btn v-btn--disabled v-btn--has-bg theme--light v-size--default\\" size=\\"x-large\\" name=\\"submit\\"><span class=\\"v-btn__content\\">
                                                Login
                                            </span></button>
                    </form>
                  </div>
                </div>
              </div>
            </div>
          </div>
        </div>
      </div>
    </div>
  </div>
</div>"
`;
